src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -609,6 +615,30 @@ bool OnOffServer::offWithEffectCommand(app::CommandHandler * commandObj, const a
     chip::EndpointId endpoint = commandPath.mEndpointId;
     Status status             = Status::Success;
 
+    if (isKnownEnumValue(effectId))
+    {
+        // Depending on effectId value, effectVariant enum type varies.
+        // The following check validates that effectVariant value is valid in relation to the applicable enum type.
+        // DelayedAllOffEffectVariantEnum or DyingLightEffectVariantEnum
+        if ((effectId == EffectIdentifierEnum::kDelayedAllOff &&
+             !isKnownEnumValue(static_cast<DelayedAllOffEffectVariantEnum>(effectVariant))) ||
+            (effectId == EffectIdentifierEnum::kDyingLight &&
+             !isKnownEnumValue(static_cast<DyingLightEffectVariantEnum>(effectVariant))))
+        {
+            status = Status::ConstraintError;
```

Comment:
Spec says:

> If the server does not support the given variant, it SHALL use the default variant.

not to error out...

Reply:
But this isn't an unsupported variant by the server but rather it is an undefined value per spec.

Reply:
It's an undefined value per spec _at the point when the server was created_, right?  Or it might be a newer spec-defined thing the server does not support.  Depends on what "support" means in that spec section.

I think the intent of the thing here is that if a new variant is added in the future clients that try to use it with old servers get the default variant instead of failures.  I agree it would be clearer if the spec said that an unsupported or unknown value is treated as default...

Reply:
Trying to get clarity on this with Rob 
https://github.com/project-chip/connectedhomeip/issues/34214#issuecomment-2237327207

I'll change if needed.